set.mm

```diff
@@ -33676,8 +33685,21 @@ sethood antecedent (Revised by BJ, 12-Feb-2022.)  Avoid ~ ax-12 .
       DODEQZUEABCUHUEEIUDJADEUDRAESTFUAUBUC $.
 
     $( Universal quantification over a restricted class abstraction.
-       (Contributed by Mario Carneiro, 3-Sep-2015.) $)
+       (Contributed by Mario Carneiro, 3-Sep-2015.)  Avoid ~ ax-8 .  (Revised
+       by Wolf Lammen, 22-Sep-2024.) $)
     ralrab2 $p |- ( A. x e. { y e. A | ph } ps <-> A. y e. A ( ph -> ch ) ) $=
+      ( crab wral cv wcel wa cab wi wal wceq wb df-rab eleq2w2 df-ral 3bitr4g
+      imbi1d albidv ax-mp ralab2 impexp albii bitr4i 3bitri ) BDAEFHZIZBDEJFKZA
+      LZEMZIZUMCNZEOZACNZEFIZUJUNPZUKUOQAEFRUTDJZUJKZBNZDOVAUNKZBNZDOUKUOUTVCVE
+      DUTVBVDBDUJUNSUBUCBDUJTBDUNTUAUDUMBCDEGUEUQULURNZEOUSUPVFEULACUFUGUREFTUH
+      UI $.
+    $( $j usage 'ralrab2' avoids 'ax-8'; $)
```

Comment:
raleqbi1dvALT is not used in the proof!

Reply:
It is inlined

Reply:
"discouraged" does not mean "forbidden". I think ~raleqbi1dvALT can and should be used here. Alternatively, we could rename ~raleqbi1dvALT, maybe ~raleqbi1dvax12, avoiding the "(New usage is discouraged.)" tag.

Reply:
I haven't seen an ALT theorem been used in a proof other than in another ALT theorem, nor a theorem proof in 2 versions without a preference for one, degrading the other clearly as FYI only.  Lets see whether others have an opinion here, too

Reply:
I've used an ALT theorem in this way before: https://us.metamath.org/mpeuni/1t1e1ALT.html

Reply:
You can add
```
  ${
    $d ph x $.  $d A x $.  $d B x $.
    raleqbidvv.1 $e |- ( ph -> A = B ) $.
    raleqbidvv.2 $e |- ( ph -> ( ps <-> ch ) ) $.
    $( Version of ~ raleqbidv with additional disjoint variable conditions, not
       requiring ~ ax-8 nor ~ df-clel .  (Contributed by BJ, 22-Sep-2024.) $)
    raleqbidvv $p |- ( ph -> ( A. x e. A ps <-> A. x e. B ch ) ) $=
      ( cv wcel wi wal wral wb wa alrimiv wceq dfcleq sylib df-ral 19.26 imbi12
      sylanbrc impcom sylg albi syl 3bitr4g ) ADIZEJZBKZDLZUIFJZCKZDLZBDEMCDFMA
      UKUNNZDLULUONABCNZUJUMNZOZUPDAUQDLURDLZUSDLAUQDHPAEFQUTGDEFRSUQURDUAUCURU
      QUPUJUMBCUBUDUEUKUNDUFUGBDETCDFTUH $.
  $}
```
which gives
```
!SHOW TRACE_BACK raleqbidvv / MATCH ax-*,df-cl*
The proof of statement "raleqbidvv" uses the following earlier statements:
  ax-mp($a) ax-1($a) ax-2($a) ax-3($a) ax-gen($a) ax-4($a) ax-5($a) ax-6($a)
  ax-7($a) ax-9($a) ax-ext($a) df-cleq($a)
```
Then use it in raleqbi1dv, so no need for an ALT version.